src/app/core/components/expansion-panel/expansion-panel.component.ts

```diff
@@ -0,0 +1,97 @@
+import { IStyles } from 'src/app/shared/models/i-styles';
+
+
+@Component({
+  selector: 'app-expansion-panel',
+  templateUrl: './expansion-panel.component.html',
+  styleUrls: [ './expansion-panel.component.scss' ]
+})
+export class ExpansionPanelComponent implements OnInit {
+  panelOpenState = false;
+  @Input() element: NameValueInterface;
+  public form: FormGroup;
+  styles: IStyles = {};
+
+  constructor(
+    private formBuilder: FormBuilder,
+    private store: Store<State>
+  ) {
+  }
+
+  ngOnInit(): void {
+    this.buildForm();
+    this.styles = this.element.styles ? JSON.parse(JSON.stringify(this.element.styles)) : this.initStyles();
+    Object.keys(this.styles).forEach(key => {
+      this.form.get(key).patchValue(this.styles[key].value);
+    });
+    this.form.get('required').patchValue(this.element.required ? this.element.required : false);
+  }
+
+  private buildForm(): void {
+    this.form = this.formBuilder.group({
+      width: [ '', [ Validators.required, this.isNumberValidator ] ],
+      height: [ '', [ Validators.required, this.isNumberValidator ] ],
+      border: [ '', [ Validators.required, this.isNumberValidator ] ],
+      borderStyle: [ '' ],
+      borderColor: [ '' ],
+      borderRadius: [ '', [ Validators.required, this.isNumberValidator ] ],
+      fontWeight: [ '', [ Validators.required, this.isNumberValidator ] ],
+      fontSize: [ '', [ Validators.required, this.isNumberValidator ] ],
+      margin: [ '', [ Validators.required, this.isNumberValidator ] ],
+      padding: [ '', [ Validators.required, this.isNumberValidator ] ],
+      label: [ '' ],
+      placeholder: [ '' ],
+      color: [ '' ],
+      required: []
+    });
+  }
+
+  public isNumberValidator = (control: FormControl): object => {
+    const condition = typeof parseInt(control.value, 10) === 'number';
+    return !condition ? { isNumberValidator: 'the value must be a number' } : null;
+  }
+
+  public OnSubmit(): void {
+    Object.keys(this.form.controls).forEach((key, index) => {
+      const controll = this.form.get(key);
+      if (controll.value && key !== 'required') {
+        this.styles[key].value = this.styles[key]?.value ? controll.value : null;
+      }
+    });
+    this.store.dispatch(actions.updateFormItemById({
+      payload: { ...this.element, styles: this.styles, required: this.form.get('required').value }
+    }));
+  }
+
+  public initStyles(): void {
+    switch (this.element.name) {
+      case EElementNames.input:
+        return JSON.parse(JSON.stringify(INPUT_STYLES));
+      case EElementNames.button:
+        return JSON.parse(JSON.stringify(BUTTON_STYLES));
+      case EElementNames.checkbox:
+        return JSON.parse(JSON.stringify(CHECKBOX_STYLES));
+      case EElementNames.select:
+        return JSON.parse(JSON.stringify(SELECT_STYLES));
+      case EElementNames.textarea:
+        return JSON.parse(JSON.stringify(TEXTAREA_STYLES));
+    }
+  }
```

Comment:
try to create a better solution
public initStyles(): void {
   return JSON.parse(JSON.stringify(styles[this.element.name]))
}